lib/cache/add-local.js

```diff
@@ -92,13 +93,21 @@ function addLocalDirectory (p, pkgData, shasum, cb) {
     getCacheStat(function (er, cs) {
       mkdir(path.dirname(pj), function (er, made) {
         if (er) return cb(er)
-        var doPrePublish = !pathIsInside(p, npm.tmp)
-        if (doPrePublish) {
-          lifecycle(data, 'prepublish', p, iferr(cb, thenPack))
-        } else {
-          thenPack()
+        var isProduction = npm.config.get('production')
+        var dir = npm.dir
+        var prefix = npm.prefix
+
+        npm.config.set('production', false)
```

Comment:
By "this stuff", I mean, trying to get/set things from the global config object, when who knows what else is running and could be effected by this.  To land this the `production` flag needs to be isolated to the installer object, so that we can instantiate an object here with the config we need and run that, rather than calling `install`.